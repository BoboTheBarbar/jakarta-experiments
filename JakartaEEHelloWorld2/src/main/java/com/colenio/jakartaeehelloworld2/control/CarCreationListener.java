package com.colenio.jakartaeehelloworld2.control;

import com.colenio.jakartaeehelloworld2.entity.CarCreated;
import jakarta.enterprise.event.Observes;

public class CarCreationListener {
    public void carCreatedEvent(@Observes CarCreated carCreated) {
        System.out.println("This car was created: " + carCreated.identifier);
    }
}
