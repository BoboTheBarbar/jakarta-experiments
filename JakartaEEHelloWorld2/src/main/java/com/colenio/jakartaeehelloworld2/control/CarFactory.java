package com.colenio.jakartaeehelloworld2.control;

import com.colenio.jakartaeehelloworld2.entity.Car;
import com.colenio.jakartaeehelloworld2.entity.Specifcation;

public class CarFactory {
    public Car createCar(Specifcation specifcation) {
        Car car = new Car();
        car.setIdentifier("Renault Corsa");
        car.setColor(specifcation.getColor());
        car.setEngineType(specifcation.getEngineType());
        return car;
    }
}
