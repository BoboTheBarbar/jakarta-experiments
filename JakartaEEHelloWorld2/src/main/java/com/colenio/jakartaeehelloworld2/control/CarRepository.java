package com.colenio.jakartaeehelloworld2.control;

import com.colenio.jakartaeehelloworld2.entity.Car;

public class CarRepository {
    public void save(Car car) {
        // This method is empty, since we don't have persistence yet.
    }
}
