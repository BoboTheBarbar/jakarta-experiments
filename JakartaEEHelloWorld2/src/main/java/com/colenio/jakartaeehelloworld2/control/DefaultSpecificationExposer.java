package com.colenio.jakartaeehelloworld2.control;

import com.colenio.jakartaeehelloworld2.entity.Color;
import com.colenio.jakartaeehelloworld2.entity.EngineType;
import com.colenio.jakartaeehelloworld2.entity.Specifcation;
import jakarta.enterprise.inject.Produces;

public class DefaultSpecificationExposer {

    @Produces
    public static final Specifcation blackElectricSuperCar = new Specifcation(Color.BLACK, EngineType.ELECTRIC);

    private DefaultSpecificationExposer() {
    }
}
