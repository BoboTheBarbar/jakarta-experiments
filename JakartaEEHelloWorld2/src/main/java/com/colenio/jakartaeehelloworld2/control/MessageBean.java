package com.colenio.jakartaeehelloworld2.control;

import jakarta.ejb.Stateless;

@Stateless
public class MessageBean {
    public String getMessage() {
        return "Hello bean!";
    }
}
