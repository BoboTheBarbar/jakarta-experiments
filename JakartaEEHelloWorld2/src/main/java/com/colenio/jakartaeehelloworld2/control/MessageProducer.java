package com.colenio.jakartaeehelloworld2.control;

import com.colenio.jakartaeehelloworld2.entity.Message;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Produces;

@ApplicationScoped
public class MessageProducer {
    @Produces
    public Message myMessage() {
        return new Message("Hello producer!");
    }
}
