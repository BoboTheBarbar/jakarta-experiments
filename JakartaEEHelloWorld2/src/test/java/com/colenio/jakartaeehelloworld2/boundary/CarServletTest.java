package com.colenio.jakartaeehelloworld2.boundary;

import com.colenio.jakartaeehelloworld2.control.CarFactory;
import com.colenio.jakartaeehelloworld2.control.CarRepository;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

class CarServletTest {

    private final CarRepository repository = new CarRepository();
    private final CarFactory factory = new CarFactory();
    CarManufacturer manufacturer = new CarManufacturer(factory, repository);


    CarServlet carServlet = new CarServlet(manufacturer);

    @Test
    void getCarHTML() {
        String actual = carServlet.getCarHTML();
        Approvals.verify(actual);
    }
}