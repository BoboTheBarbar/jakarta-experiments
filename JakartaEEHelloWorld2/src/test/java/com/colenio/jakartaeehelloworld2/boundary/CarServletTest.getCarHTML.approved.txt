<html><body>
<h1>Hello! This is your car!</h1>
<h1> Renault Corsa </h1>
<h1> RED </h1>
<h1> ELECTRIC </h1>
</body></html>
